app/models/manageiq/providers/hawkular/inventory/server_operations.rb

```diff
@@ -88,23 +90,21 @@ def add_middleware_datasource(ems_ref, hash)
             end
           end
 
-          def add_middleware_deployment(ems_ref, hash)
+          def add_middleware_deployment(ems_ref, feed_id, hash)
             with_provider_connection do |connection|
               deployment_data = {
                 :enabled               => hash[:file]["enabled"],
                 :force_deploy          => hash[:file]["force_deploy"],
                 :destination_file_name => hash[:file]["runtime_name"] || hash[:file]["file"].original_filename,
                 :binary_content        => hash[:file]["file"].read,
-                :resource_path         => ems_ref.to_s
+                :resource_id           => ems_ref.to_s,
+                :feed_id               => feed_id
               }
 
               unless hash[:file]['server_groups'].nil?
                 # in case of deploying into server group the resource path should point to the domain controller
                 deployment_data[:server_groups] = hash[:file]['server_groups']
-                server_group_path_hash = ::Hawkular::Inventory::CanonicalPath.parse(deployment_data[:resource_path]).to_h
-                server_group_path_hash[:resource_ids].slice!(1..-1)
-                host_controller_path = ::Hawkular::Inventory::CanonicalPath.new(server_group_path_hash)
-                deployment_data[:resource_path] = host_controller_path.to_s
+                deployment_data[:resource_id] = connection.inventory.resource(deployment_data[:resource_id]).parent_id
```

Comment:
Shouldn't be better to use MiQ db to get the ID of the parent?
I guess you could pass the `parent_id` from the UI in `hash`.

Reply:
!!!
I forgot about that hash. Will send a PR fixing that.